tests/filecheck/transforms/csl-stencil-handle-async-flow-no-args.mlir

```diff
@@ -0,0 +1,199 @@
+// RUN: xdsl-opt %s -p "csl-stencil-handle-async-flow" | filecheck %s
+
+builtin.module {
+  "csl_wrapper.module"() <{height = 4 : i16, params = [#csl_wrapper.param<"z_dim" default=4 : i16>, #csl_wrapper.param<"pattern" default=2 : i16>, #csl_wrapper.param<"num_chunks" default=1 : i16>, #csl_wrapper.param<"chunk_size" default=2 : i16>, #csl_wrapper.param<"padded_z_dim" default=2 : i16>], program_name = "loop_kernel", width = 4 : i16}> ({
+  ^0(%arg35 : i16, %arg36 : i16, %arg37 : i16, %arg38 : i16, %arg39 : i16, %arg40 : i16, %arg41 : i16, %arg42 : i16, %arg43 : i16):
+    %0 = arith.constant 1 : i16
+    %1 = arith.constant 0 : i16
+    %2 = "csl.get_color"(%1) : (i16) -> !csl.color
+    %3 = "csl_wrapper.import"(%arg37, %arg38, %2) <{fields = ["width", "height", "LAUNCH"], module = "<memcpy/get_params>"}> : (i16, i16, !csl.color) -> !csl.imported_module
+    %4 = "csl_wrapper.import"(%arg40, %arg37, %arg38) <{fields = ["pattern", "peWidth", "peHeight"], module = "routes.csl"}> : (i16, i16, i16) -> !csl.imported_module
+    %5 = "csl.member_call"(%4, %arg35, %arg36, %arg37, %arg38, %arg40) <{field = "computeAllRoutes"}> : (!csl.imported_module, i16, i16, i16, i16, i16) -> !csl.comptime_struct
+    %6 = "csl.member_call"(%3, %arg35) <{field = "get_params"}> : (!csl.imported_module, i16) -> !csl.comptime_struct
```

Comment:
Is there any way to reduce this test case to detect exactly your change? I would recommend trying to shrink it with shrinkray by catching the out of bounds error and re-raising a ShrinkException

Reply:
That's the lazy way to do it, anyway, I'm sure it's also reducible by hand.